idaes/conftest.py

```diff
@@ -29,37 +29,72 @@
 ####
 
 ####
-# Only run the tests for your particular platform(s), if it is marked with those platform(s)
-# e.g. to mark tests for linux:
-#
-# import pytest
-# @pytest.mark.linux
-# def test_something():
-#    print("this only runs on linux")
-#
-#
-# In addition, you can use "no<platform>" to exclude
-#
-# @pytest.mark.nowin32
-# def test_something():
-#    print("this will not run on windows")
-#
-# The names of the platforms should match what is returned by `sys.platform`, in particular:
-#    Linux = 'linux'
-#    Windows = 'win32'
-#    macOS = 'darwin'
 
 
-ALL = {"darwin", "linux", "win32"}
-ALL_NO = {"no" + tag for tag in ALL}
+@pytest.hookimpl
+def pytest_runtest_setup(item):
+    _validate_required_markers(
+        item,
+        required_markers={"unit", "component", "integration"},
+        expected_count=1
+    )
+    _skip_for_unsupported_platforms(
+        item,
+        all_platforms={"darwin", "linux", "win32"},
```

Comment:
I prefer to define things like this in constants so you can change them without digging through the code.

Reply:
Makes sense. I'll move it to a top-level module constant.